services/slam/builtin/builtin.go

```diff
@@ -571,6 +571,23 @@ func (slamSvc *builtIn) GetMap(
 	return mimeType, imData, vObj, nil
 }
 
+// GetInternalState forwards the request for the SLAM algorithms's internal state. Once a response is received, it is returned
+// to the user.
+func (slamSvc *builtIn) GetInternalState(ctx context.Context, name string) ([]byte, error) {
+	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetInternalState")
```

Comment:
Just for my own edification, what do we use these spans / traces for?

Reply:
Spans and traces can be used with a -trace flag to check the time various functions take (hence the `defer span.End()`)

https://pkg.go.dev/runtime/trace